src/server/scripts/Spells/spell_monk.cpp

```diff
@@ -294,11 +297,65 @@ class spell_monk_surging_mist_glyphed : public SpellScriptLoader
     }
 };
 
+// Roll - 109132 or Roll (3 charges) - 121827
+class spell_monk_roll : public SpellScriptLoader
+{
+public:
+	spell_monk_roll() : SpellScriptLoader("spell_monk_roll") { }
+
+	class spell_monk_roll_SpellScript : public SpellScript
+	{
+		PrepareSpellScript(spell_monk_roll_SpellScript);
+
+		bool Validate(SpellInfo const* /*spell*/)
+		{
+			if (!sSpellMgr->GetSpellInfo(SPELL_MONK_ROLL))
+				return false;
+			return true;
+		}
+
+		void HandleBeforeCast()
+		{
+			Aura* aur = GetCaster()->AddAura(SPELL_MONK_ROLL_TRIGGER, GetCaster());
+			if (!aur)
+				return;
+
+			AuraApplication* app = aur->GetApplicationOfTarget(GetCaster()->GetGUID());
+			if (!app)
+				return;
+
+			app->ClientUpdate();
+		}
+
+		void HandleAfterCast()
+		{
+			Unit* caster = GetCaster();
+			if (!caster || caster->GetTypeId() != TYPEID_PLAYER)
```

Comment:
Caster cant be null in SpellScript, and this check ```caster->GetTypeId() != TYPEID_PLAYER``` should be in Load function :)